tools/firmware-utils/src/tplink-safeloader.c

```diff
@@ -2473,6 +2473,82 @@ static struct device_info boards[] = {
 		.first_sysupgrade_partition = "os-image",
 		.last_sysupgrade_partition = "file-system"
 	},
+	{
+		.id = "DECO-M5-EU-V3.0",
+		.vendor = "",
+		.support_list =
+			"SupportList:\n"
+			"{product_name:M5,product_ver:1.0.0,special_id:55530000}\n"
+			"{product_name:M5,product_ver:1.0.0,special_id:45550000}\n"
+			"{product_name:M5,product_ver:1.0.0,special_id:43410000}\n"
+			"{product_name:M5,product_ver:1.0.0,special_id:4A500000}\n"
+			"{product_name:M5,product_ver:1.0.0,special_id:41550000}\n"
+			"{product_name:M5,product_ver:1.0.0,special_id:4B520000}\n"
+			"{product_name:M5,product_ver:1.0.0,special_id:49440000}\n"
+			"{product_name:M5,product_ver:1.0.0,special_id:53570000}\n"
+			"{product_name:M5,product_ver:1.0.0,special_id:42340000}\n"
+			"{product_name:M5,product_ver:3.0.0,special_id:53570000}\n"
+			"{product_name:M5,product_ver:3.0.0,special_id:42340000}\n"
+			"{product_name:M5,product_ver:3.0.0,special_id:54570000}\n"
+			"{product_name:M5,product_ver:3.0.0,special_id:55530000}\n"
+			"{product_name:M5,product_ver:3.0.0,special_id:45550000}\n"
+			"{product_name:M5,product_ver:3.0.0,special_id:43410000}\n"
+			"{product_name:M5,product_ver:3.0.0,special_id:4A500000}\n"
+			"{product_name:M5,product_ver:3.0.0,special_id:41550000}\n"
+			"{product_name:M5,product_ver:3.0.0,special_id:4B520000}\n"
+			"{product_name:M5,product_ver:3.0.0,special_id:49440000}\n"
+			"{product_name:M5,product_ver:3.2.0,special_id:55530000}\n"
+			"{product_name:M5,product_ver:3.2.0,special_id:45550000}\n"
+			"{product_name:M5,product_ver:3.2.0,special_id:43410000}\n"
+			"{product_name:M5,product_ver:3.2.0,special_id:4A500000}\n"
+			"{product_name:M5,product_ver:3.2.0,special_id:41550000}\n"
+			"{product_name:M5,product_ver:3.2.0,special_id:4B520000}\n"
+			"{product_name:M5,product_ver:3.2.0,special_id:49440000}\n"
+			"{product_name:M5,product_ver:3.2.0,special_id:53570000}\n"
+			"{product_name:M5,product_ver:3.2.0,special_id:42340000}\n"
+			"{product_name:M5,product_ver:3.2.0,special_id:54570000}\n",
```

Comment:
Ah, so now you are contradicting yourself, since this is neither limited to 3.0 nor to EU.
And please do not add 1.0 strings to a v3. This should be sorted, and please check whether the partition table in all of the devices finally chosen is the same.

Reply:
Some more info about this device... V1, V3 and V3.2 seem compatible with each other. I cannot say for 100% certainty, but the stock firmware is identical for all three. It only differs on V2... with that said, I have a V3 / CA working flawlessly. These "special_id's" match what is found in stock firmware.

Reply:
v1 and v3 should be treated separately if v2 is different.